File: 백준/Silver/18258. 큐 2/큐 2.cc
#include <bits/stdc++.h>
using namespace std;

int main(void) {
	ios::sync_with_stdio(0);
	cin.tie(0);
	queue<int> Q;
	int command;
	cin >> command;

	for (int i = 0; i < command; i++) {
		string s;
		cin >> s;
		if (s == "push") {
			int a;
			cin >> a;
			Q.push(a);
		}
		else if (s == "front") {
			if (Q.empty()) cout << -1 << '\n';
			else cout << Q.front() << '\n';
		}
		else if (s == "back") {
			if (Q.empty()) cout << -1 << '\n';
			else cout << Q.back() << '\n';
		}
		else if (s == "empty") {
			if (Q.empty()) cout << 1 << '\n';
			else cout << 0 << '\n';
		}
		else if (s == "size") {
			cout << Q.size() << '\n';
		}
		else {
			if (Q.empty()) cout << -1 << '\n';
			else {
				cout << Q.front() << '\n';
				Q.pop();
			}
		}
	}
	
}

File: 백준/Silver/18258. 큐 2/큐 2.java
import java.util.*;
import java.io.*;
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        Deque<Integer> q = new LinkedList<>();
        StringTokenizer st;
        StringBuilder sb = new StringBuilder();
        for(int i=0;i<N;i++) {
            st = new StringTokenizer(br.readLine());
            String command = st.nextToken();
            int val;
            if(command.equals("push")) {
                val = Integer.parseInt(st.nextToken());
                q.add(val);
            }
            else if(command.equals("pop")) {
                if(q.isEmpty()) sb.append("-1").append("\n");
                else sb.append(q.pop()).append("\n");
            }
            else if(command.equals("front")) {
                if(q.isEmpty()) sb.append("-1").append("\n");
                else sb.append(q.getFirst()).append("\n");;
            }
            else if(command.equals("back")) {
                if(q.isEmpty()) sb.append("-1").append("\n");
                else sb.append(q.getLast()).append("\n");;
            }
            else if(command.equals("empty")) {
                if(q.isEmpty()) sb.append("1").append("\n");
                else sb.append("0").append("\n");
            }
            else if(command.equals("size")) {
                sb.append(q.size()).append("\n");
            }
        }
        System.out.println(sb);
    }
}